.gitmodules

```diff
@@ -22,3 +22,6 @@
 [submodule "external/debugger"]
 	path = external/debugger
 	url = https://github.com/kevinb7/debugger
+[submodule "external/p5"]
+	path = external/p5
+	url = https://github.com/kevinb7/p5.js
```

Comment:
For sure.  This is a huge change.  I found a bug last relating to how it implements "color".  This whole pull request is more an experiment.  I made a it a pull request so you and the team could see various changes that we may want to do in the future.  I started out actually pull stuff out of processing-js like the Java parser and all the 3D code and then I though it might be easier to integrate p5.js (and you had mentioned something about doing that a while ago) so I thought I'd give it a try.